src/core/components/no-keychain.js

```diff
@@ -0,0 +1,24 @@
+'use strict'
+
+function fail () {
+  throw new Error('Key management requires \'--pass ...\' option')
+}
+
+class NoKeychain {
+  static get options () { fail() }
+  static generateOptions () { fail() }
+
+  createKey () { fail() }
+  listKeys () { fail() }
+  findKeyById () { fail() }
+  findKeyByName () { fail() }
+  renameKey () { fail() }
+  removeKey () { fail() }
+  exportKey () { fail() }
+  importKey () { fail() }
+  importPeer () { fail() }
+
+  get cms () { fail() }
+}
+
+module.exports = NoKeychain
```

Comment:
Is this still needed ??

Reply:
Yes, we need it.  If the IPFS is started without `--pass`, then `no-keychain` is used.

Reply:
Isn't `pass` just the key to encrypt it locally? The user should be able to run all of these methods even if locally the key is stored in plain text.

Reply:
No key is stored in plain text.  All keys are encrypted at rest with an encryption key that is derived from the the pass phrase `--pass`.  See https://github.com/libp2p/js-libp2p-keychain#private-key-storage for more details.

Reply:
Why can't the private keys be stored as plaintext and within reach from the local user without this password-PBKDF2-whatever stuff? `go-ipfs` does that.